### PR TITLE
Feature: add the output of E_KS(sigma->0)

### LINKED ISSUE #2556: Add print of scf-calculated energy(sigma->0）
### Background

When use abacus to do scf-calculation in `nspin=2` setting, I cannot find the print-out of the real electronic energy.
which can be easily read in VASP by `energy(sigma->0)` in `OUTCAR` and `E0` in `OSZICAR`
Now I can only use `ETOT` minus `E_demet` to get the energy, which only means `energy without entropy` in `OUTCAR`

### Describe the solution you'd like

1. Give a print-out of `energy(sigma->0)` in `running_scf.log` will be a direct help
2. the log file for abacus calculation seems to be hard to read, many k-points information is added in, which users are not usually care about, could a document about how to read the output file of abacus in a highly-efficient way be written?

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Chentao168**
1. Add the output of `E_KS(sigma->0)`, which is the same as `energy(sigma->0)` in the VASP.
image
image
Here, I calculate `E_KS(sig->0) = (this->f_en.etot - this->f_en.demet/(2+n_order))` .
I find `n_order = std::max(0, Occupy::gaussian_type)`.
The following pictures are the related codes from VASP:
image
image
2. Rename `E_demet` to `E_entropy(-TS)`.
3. Change `setw(12)` to `setw(16)` for allowing to show the `E_KS(sigma->0)` and `E_entropy(-TS)`.

### CODE REVIEW

File: source/module_elecstate/elecstate_print.cpp [hunk 1]
```diff
@@ -173,16 +174,18 @@ void ElecState::print_etot(const bool converged,
...
         this->print_format("E_KohnSham", this->f_en.etot);
+        this->print_format("E_KS(sig->0)", this->f_en.etot - this->f_en.demet/(2+n_order));
         this->print_format("E_Harris", this->f_en.etot_harris);
         this->print_format("E_band", this->f_en.eband);
         this->print_format("E_one_elec", this->f_en.eband + this->f_en.deband);
         this->print_format("E_Hartree", this->f_en.hartree_energy);
         this->print_format("E_xc", this->f_en.etxc - this->f_en.etxcc);
         this->print_format("E_Ewald", this->f_en.ewald_energy);
-        this->print_format("E_entropy (-TS)", this->f_en.demet); // mohan add 2011-12-02
+        this->print_format("E_entro(-TS)", this->f_en.demet); // mohan add 2011-12-02
```
Comment (hongriTianqi) [inline source/module_elecstate/elecstate_print.cpp]: E_entro is more unclear than E_entropy. Is there any special reason for this change of naming?

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

**Comment by wenfei-li** [comment]
how does the content of the pr relate to the title? I only saw change in E_entropy not E_KS(sig->0)

File: source/module_elecstate/fp_energy.cpp [hunk 2]
```diff
@@ -40,7 +40,7 @@ void fenergy::print_all() const
     std::cout << " etxc-etxcc=" << etxc - etxcc << std::endl;
     std::cout << " ewld=" << ewald_energy << std::endl;
     std::cout << " ehart=" << hartree_energy << std::endl;
-    std::cout << " demet=" << demet << std::endl;
+    std::cout << " entropy(-TS)=" << demet << std::endl;
```
Comment (hongriTianqi) [inline source/module_elecstate/fp_energy.cpp:43]: Please add a unit test for this function.

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

File: source/module_elecstate/test/elecstate_fp_energy_test.cpp [hunk 3]
```diff
@@ -55,6 +55,7 @@ TEST_F(fenergy, print_all)
     f_en.print_all();
     std::string output = testing::internal::GetCapturedStdout();
     EXPECT_THAT(output, testing::HasSubstr("total="));
+    EXPECT_THAT(output, testing::HasSubstr("entropy(-TS)="));
```
Comment (hongriTianqi) [inline source/module_elecstate/test/elecstate_fp_energy_test.cpp:58]: It is confusing to use entropy at one place, but entro at another place. Could you please give comments about their difference in the code?

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

**Review by hongriTianqi — CHANGES_REQUESTED** [review]
Add necessary explanation for your pr

File: source/module_elecstate/elecstate_print.cpp [hunk 4]
```diff
@@ -173,16 +174,18 @@ void ElecState::print_etot(const bool converged,
 
     if (printe > 0 && ((iter + 1) % printe == 0 || converged || iter == GlobalV::SCF_NMAX))
     {
+        int n_order = std::max(0, Occupy::gaussian_type);
         GlobalV::ofs_running << "\n " << std::setw(12) << "Energy" << std::setw(30) << "Rydberg" << std::setw(30)
                              << "eV" << std::endl;
         this->print_format("E_KohnSham", this->f_en.etot);
+        this->print_format("E_KS(sig->0)", this->f_en.etot - this->f_en.demet/(2+n_order));
```
Comment (dyzheng) [inline source/module_elecstate/elecstate_print.cpp]: @wenfei-li this line add E(sigma->0), but @Chentao168 can you add some annotation for this line? what is the related formula?

File: source/module_elecstate/elecstate_print.cpp [hunk 5]
```diff
@@ -173,16 +174,18 @@ void ElecState::print_etot(const bool converged,
 
     if (printe > 0 && ((iter + 1) % printe == 0 || converged || iter == GlobalV::SCF_NMAX))
     {
+        int n_order = std::max(0, Occupy::gaussian_type);
         GlobalV::ofs_running << "\n " << std::setw(12) << "Energy" << std::setw(30) << "Rydberg" << std::setw(30)
```
Comment (hongriTianqi) [inline source/module_elecstate/elecstate_print.cpp]: please align all the energy terms.

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

**Review by hongriTianqi — APPROVED** [review]

**Review by dyzheng — APPROVED** [review]

**Pull request merged**